cmsis_rtos_v1: add semaphore negative tests, replace an else case

nniranjhana opened the pull request:
Added couple of negative tests for osSemaphoreWait
and osSemaphoreRelease to increase code coverage.

Replaced an else if case in osSemaphoreWait with else
to account for both EBUSY and EAGAIN return values
from k_sem_take. The return value would be 0 for
osSemaphoreWait in both cases.

Commit:
tests: cmsis_rtos_v1: add semaphore negative tests

Added couple of negative tests for osSemaphoreWait
and osSemaphoreRelease to increase code coverage.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

rgundi commented on lib/cmsis_rtos_v1/cmsis_semaphore.c:
```diff
@@ -68,7 +68,7 @@ int32_t osSemaphoreWait(osSemaphoreId semaphore_id, uint32_t timeout)
+	} else {
 		return 0;
 	}
 
```
remove the below return statement as that'll not be executed now.

rgundi commented on tests/cmsis_rtos_v1/src/semaphore.c:
```diff
@@ -16,6 +16,10 @@ void thread_sema(void const *arg)
+	/* Try taking when no semaphore is available */
+	zassert_false(osSemaphoreWait((osSemaphoreId)arg, 0),
+			"Something's wrong with osSemaphoreWait!");
+
```
Please modify the statement above to make it similar to the below statements. Also, change the comment to indicate this is a testcase with 0 timeout.

nniranjhana commented:
Hi @rgundi addressed your review comments, thanks

Commit:
lib: cmsis_rtos_v1: replace an else case

Replace an else-if case in osSemaphoreWait with
else to account for both EBUSY and EAGAIN return
values from k_sem_take. The return value would be
0 for osSemaphoreWait in both cases.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

rgundi reviewed (approved):
LGTM
